source/gameengine/GamePlayer/GPG_ghost.cpp

```diff
@@ -582,6 +589,40 @@ static bool quitGame(KX_ExitRequest exitcode)
 	return (exitcode != KX_ExitRequest::RESTART_GAME && exitcode != KX_ExitRequest::START_OTHER_GAME);
 }
 
+#ifdef WITH_GAMEENGINE_BPPLAYER
+
+static BlendFileData *load_encrypted_game_data(const char *filename, std::string localPath, std::string encryptKey)
+{
+	ReportList reports;
+	BlendFileData *bfd = NULL;
+	char *fileData = NULL;
+	int fileSize;
+	BKE_reports_init(&reports, RPT_STORE);
+
+	if (filename == NULL) {
+		return NULL;
+	}
+
+	if (!localPath.empty() && !encryptKey.empty()) {
+		// Load file and decrypt.
+		fileData = SPINDLE_DecryptFromFile(filename, &fileSize, encryptKey.c_str(), 0);
+	}
+
+	if (fileData) {
+		bfd = BLO_read_from_memory(fileData, fileSize, localPath.c_str(), &reports, BLO_READ_SKIP_USERDEF);
```

Comment:
The usage of the `relabase` / `localPath` could not be made by calling a setter of this value instead adding an argument to the read/open file functions ?